### ISSUE TITLE
fix(pytorch): in-place view mutations in SafeSVD/SafeEigh backward block higher-order gradients and torch.compile

### ISSUE THREAD

**Issue by hunter-heidenreich** [hunter-heidenreich/Kabsch-Cookbook#111]
## Problem

Both `SafeSVD.backward` and `SafeEigh.backward` use `.diagonal(dim1=-2, dim2=-1).fill_(1.0)` and `.diagonal(dim1=-2, dim2=-1).zero_()` to set diagonal values. `.diagonal()` returns a view, so `.fill_()` and `.zero_()` are in-place mutations through a view.

**Locations:**
- `src/kabsch_horn/pytorch/kabsch_svd_nd.py:73-77`
- `src/kabsch_horn/pytorch/horn_quat_3d.py:33-35`

## Impact

1. **Higher-order gradients**: If a user calls with `create_graph=True` (for Hessian-vector products, meta-learning, etc.), these in-place mutations will raise a `RuntimeError` or silently produce incorrect higher-order gradients, because PyTorch's autograd cannot track in-place ops on graph intermediates when building a higher-order graph.

2. **torch.compile**: These patterns are a known source of graph breaks under `torch.compile` (Dynamo + Inductor). In-place mutation through views forces the compiler to insert a graph break and fall back to eager execution.

## Suggested Fix

Replace in-place view mutations with out-of-place operations:

```python
eye = torch.eye(safe_D.shape[-1], dtype=safe_D.dtype, device=safe_D.device)
safe_D = torch.where(eye.bool(), torch.ones_like(safe_D), safe_D)
F = 1.0 / safe_D
F = torch.where(eye.bool(), torch.zeros_like(F), F)
```

Found by automated audit.
